### PR TITLE
Add Cypress reporter

### PR DESCRIPTION
**Pull request by nprizal**
### Description
Adds Cypress reporter that extends Mocha reporter. Cypress built on top of Mocha, so there's lots of overlap with the Mocha reporter, except for how it parses errors.

This PR is made on top of the amazing work started by @jesseduffield in https://github.com/buildkite/test-collector-javascript/pull/47

### Changes

- Add Cypress reporter
- Add examples and e2e test for Cypress reporter
- Update pipeline to use cypress base image
- Update docs

### CODE REVIEW

**Comment by nprizal** [comment]
Hey @jesseduffield! I fix the CI pipeline, and it's now working 🎉. I also DRY-ed up the reporter to extends the mocha reporter. Can you have a look and give it a test?

File: examples/cypress/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+						presets: ['@babel/preset-env', '@babel/preset-react'],
+					},
+				},
+			},
+		],
+	},
+	resolve: {
+		extensions: ['.js', '.jsx'],
+	},
+};
```
Comment (meghan-kradolfer) [inline examples/cypress/webpack.config.js:19]: Not sure if the indenting on these files is really large or if it's just github..

File: util/failureExpanded.js [hunk 2]
```diff
@@ -8,4 +8,4 @@ const failureExpanded = (errors) => {
   })
 }
 
-module.exports = failureExpanded
\ No newline at end of file
+module.exports = failureExpanded
```
Comment (meghan-kradolfer) [inline util/failureExpanded.js:11]: Noticed a few of these files have the newline at end of file removed. I think this is a pattern to have them in BK, so maybe we should keep them there?

Comment (nprizal) [reply util/failureExpanded.js:11]: This PR actually adds new line, that wasn't there originally. But I'll reset the changes, as it's not related. We need to setup linter for test-collector-javascript

File: e2e/cypress.test.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    ...process.env,
+    BUILDKITE_ANALYTICS_TOKEN: "xyz",
+    BUILDKITE_ANALYTICS_CYPRESS_TOKEN: "abc",
+    BUILDKITE_ANALYTICS_DEBUG_ENABLED: "true",
+  }
+
+  describe('when token is defined through config', () => {
+    test('it uses the correct token', (done) => {
+      exec('npm test -- --config-file cypress.config.token.js',
+        { cwd, env: { ...env, BUILDKITE_ANALYTICS_TOKEN: undefined } }, (error, stdout, stderr) => {
```
[the remark this answers is not in the document]
Comment (meghan-kradolfer) [reply e2e/cypress.test.js:22]: Makes sense! Thanks 😊

**Review by meghan-kradolfer — APPROVED** [review]

**Comment by nprizal** [comment]
resolves https://github.com/buildkite/test-collector-javascript/issues/13

**Pull request merged**